Realm/RLMObjectStore.mm

```diff
@@ -87,6 +87,34 @@ static void RLMVerifyAndAlignColumns(RLMObjectSchema *tableSchema, RLMObjectSche
     objectSchema.properties = properties;
 }
 
+// ensure all search indexes for all tables are up-to-date
+// does not need to be called from a write transaction
+static void RLMRealmUpdateIndexes(RLMRealm *realm) {
+    bool commitWriteTransaction = false;
+    for (RLMObjectSchema *objectSchema in realm.schema.objectSchema) {
+        realm::Table *table = objectSchema.table;
+        for (RLMProperty *prop in objectSchema.properties) {
+            // FIXME: support other types
+            if (prop.type == RLMPropertyTypeString && prop.indexed != table->has_search_index(prop.column)) {
+                if (!realm.inWriteTransaction) {
+                    [realm beginWriteTransaction];
+                    commitWriteTransaction = true;
+                }
+                if (prop.indexed) {
+                    table->add_search_index(prop.column);
+                }
+                else {
+                    table->remove_search_index(prop.column);
```

Comment:
The check is there only because the old version of the code had a similar check, but it does appear to be pointless as indexing non-string columns is rejected earlier.